### PR TITLE
Implemented my enumerable methods

### PR DESCRIPTION
**Pull request by elshaka**

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] returns an Enumerator if no block is given. You can achieve this by using ```return to_enum```

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] returns an enumerator if no block is given

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_select
+    selected = []
+    my_each do |i|
+      selected << i if yield(i)
+    end
+    selected
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] returns an enumerator if no block is given

File: my_enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_all?
+    my_each do |i|
+      return false unless yield(i)
+    end
+    true
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] when no block or argument is given returns true when none of the collection members are false or nil
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: my_enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_any?
+    my_each do |i|
+      return true if yield(i)
+    end
+    false
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: my_enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_none?(&block)
+    !my_any?(&block)
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: my_enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_count
+    count = 0
+    count += 1 while self[count]
+    count
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] counts the number of items in enum that are equal to item if an argument is given
- [ ] counts the number of elements yielding a true value if a block is given

File: my_enumerable.rb [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  def my_inject(initial = nil)
+    list = self
+    list = [initial] + list if initial
+    return nil if list.empty?
+
+    result = list[0]
+    list[1..-1].my_each { |i| result = yield(result, i) }
+    result
+  end
```
Comment (whiz25) [inline my_enumerable.rb]: - [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

File: my_enumerable.rb [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      initial = args[0]
+    end
+    list = self
+    list = [initial] + list if initial
+    return nil if list.empty?
+
+    result = list[0]
+    list[1..-1].my_each { |i| result = method ? result.send(method, i) : yield(result, i) }
+    result
+  end
```
Comment (wathigo) [inline my_enumerable.rb]: **[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
     
     TypeError:
       no implicit conversion of Range into Array
```
* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
```
Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)
     
     TypeError:
       no implicit conversion of Range into Array
```

**Pull request merged**